portal-bridge/src/types/state/execution.rs

```diff
@@ -40,35 +42,43 @@ struct GenesisConfig {
     state_root: B256,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct AccountProof {
-    pub address: Address,
-    pub state: AccountStateInfo,
-    pub proof: Vec<Bytes>,
-}
-
 pub struct State {
     pub database: CacheDB<EmptyDB>,
     next_block_number: u64,
 }
 
 const GENESIS_STATE_FILE: &str = "resources/genesis/mainnet.json";
 
+impl Default for State {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl State {
-    pub fn new() -> anyhow::Result<Self> {
-        let mut cache_db = CacheDB::new(EmptyDB::default());
-        let mut accounts = BTreeSet::new();
+    pub fn new() -> Self {
+        State {
+            next_block_number: 0,
+            database: CacheDB::new(EmptyDB::default()),
+        }
+    }
+
+    pub fn initialize_genesis(&mut self) -> anyhow::Result<RootWithTrieDiff> {
+        ensure!(
+            self.next_block_number == 0,
+            "Expected block {}, received 0",
```

Comment:
nit: This error is a bit confusing. Maybe something like: `Trying to initialize genesis but expecting block {}`.